### ISSUE TITLE
Manually typing an invalid date, then a valid date, removes timezone

### ISSUE THREAD

**Issue by Colkadome** [YouCanBookMe/react-datetime#318]
### I'm Submitting a ...
```
[ x] Bug report
[ ] Feature request
[ ] Support request
```

### Steps to Reproduce
1. Create a new moment with a timezone, e.g. moment.tz( . . . , 'Australia/Melbourne')
2. Have a Datetime controlled component that takes the moment as a value, and updates the value with onChange.
3. Type in an invalid date.
4. Type in a valid date.

### Expected Results
The timezone is kept.

### Actual Results
The returned moment has no timezone set (e.g. myMoment.tz() returns undefined).

### Other Information (e.g. stacktraces, related issues, suggestions how to fix)
This only happens when you edit the input value manually. If you use the picker, the bug doesn't occur. Right now I am fixing this with the following code in my React component (ES6):

```
onChange(value) {
  
  if (typeof(value) === 'string') {

    this.setState({ value });

  } else {

    this.setState({ value: moment.tz(value.format('YYYY-MM-DD[T]HH:mm:ss'), 'Australia/Melbourne') });
  }

}
```
Basically I'm removing the UTC offset, then adding the timezone. It could be cool to have a "tz" prop for the Datetime component that somehow preserves the timezone.

**Comment by simeg** [comment]
Closing due to inactivity, let me know if you want it re-opened.
